fe/fe-core/src/main/java/com/starrocks/analysis/CreateRoutineLoadStmt.java

```diff
@@ -285,6 +297,33 @@ public void analyze(Analyzer analyzer) throws UserException {
         checkDataSourceProperties();
     }
 
+    public static RoutineLoadDesc getLoadDesc(OriginStatement origStmt, Map<String, String> sessionVariables) {
+        long sqlMode;
+        if (sessionVariables != null && sessionVariables.containsKey(SessionVariable.SQL_MODE)) {
+            sqlMode = Long.parseLong(sessionVariables.get(SessionVariable.SQL_MODE));
+        } else {
+            sqlMode = SqlModeHelper.MODE_DEFAULT;
+        }
+
+        // parse the origin stmt to get routine load desc
+        SqlParser parser = new SqlParser(new SqlScanner(new StringReader(origStmt.originStmt), sqlMode));
+        try {
+            StatementBase stmt = SqlParserUtils.getStmt(parser, origStmt.idx);
+            if (stmt instanceof CreateRoutineLoadStmt) {
+                return CreateRoutineLoadStmt.
+                        buildLoadDesc(((CreateRoutineLoadStmt) stmt).getLoadPropertyList());
+            } else if (stmt instanceof AlterRoutineLoadStmt) {
+                return CreateRoutineLoadStmt.
+                        buildLoadDesc(((AlterRoutineLoadStmt) stmt).getLoadPropertyList());
+            } else {
+                throw new IOException("stmt is neither CreateRoutineLoadStmt nor AlterRoutineLoadStmt");
+            }
+        } catch (Exception e) {
+            LOG.error("error happens when parsing create/alter routine load stmt: " + origStmt.originStmt, e);
+            return null;
```

Comment:
why return null not throw exception, inconsistent with previous behavior

Reply:
If exception is thrown here, the fe will start failed.  I'm afraid that there may be some corner case I have not considered. This is not a critical problem, so we just return null, and the load desc can be null.